engine/engine/src/engine.cpp

```diff
@@ -411,6 +411,25 @@ namespace dmEngine
         delete engine;
     }
 
+    dmPlatform::PlatformOpenGLVersion ConvertOpenGLVersionHint(const char* version)
+    {
+    #define CHECK_AND_RETURN_OPENGL_VERSION(version_str, version_enum) \
+        if (strcmp(version, #version_str) == 0) return dmPlatform::PLATFORM_OPENGL_VERSION_##version_enum
+
+        CHECK_AND_RETURN_OPENGL_VERSION(opengl_3_3, 3_3);
+        CHECK_AND_RETURN_OPENGL_VERSION(opengl_4_0, 4_0);
+        CHECK_AND_RETURN_OPENGL_VERSION(opengl_4_1, 4_1);
+        CHECK_AND_RETURN_OPENGL_VERSION(opengl_4_2, 4_2);
+        CHECK_AND_RETURN_OPENGL_VERSION(opengl_4_3, 4_3);
+        CHECK_AND_RETURN_OPENGL_VERSION(opengl_4_4, 4_4);
+        CHECK_AND_RETURN_OPENGL_VERSION(opengl_4_5, 4_5);
+        CHECK_AND_RETURN_OPENGL_VERSION(opengl_4_6, 4_6);
+        CHECK_AND_RETURN_OPENGL_VERSION(opengl_highest_available, HIGHEST_AVAILABLE);
```

Comment:
I think an encoded int should suffice here, with e.g. 0 being the highest available.